x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/AuthorizationEngine.java

```diff
@@ -667,6 +697,103 @@ public IndicesAccessControl getIndicesAccessControl() {
         }
     }
 
+    /**
+     * Holds information about authorization of a parent action which is used to pre-authorize its child actions.
+     *
+     *  @param action the parent action
+     */
+    record ParentActionAuthorization(String action) implements Writeable, ToXContent {
+
+        public static final String THREAD_CONTEXT_KEY = "_xpack_security_authorization";
```

Comment:
nit: rename to "_xpack_security_parent_action_authz"

Reply:
Good suggestion. I've renamed it https://github.com/elastic/elasticsearch/pull/91886/commits/95e1054a394274190a1f12dc683a62e779dc8b5f